indico/modules/payment/forms.py

```diff
@@ -64,26 +63,8 @@ def validate_currency(self, field):
 
 
 class EventSettingsForm(IndicoForm):
-    currency = SelectField(_('Currency'), [DataRequired()])
     conditions = TextAreaField(_('Conditions'), description=CONDITIONS_DESC)
 
     def __init__(self, *args, **kwargs):
         self._event = kwargs.pop('event')
         super(EventSettingsForm, self).__init__(*args, **kwargs)
-        self._set_currencies()
-
-    def _set_currencies(self):
-        currencies = [(c['code'], '{0[code]} ({0[name]})'.format(c)) for c in settings.get('currencies')]
-        # In case the event's currency was deleted, we keep it here anyway.
-        event_currency = self.currency.object_data
-        if event_currency and not any(x[0] == event_currency for x in currencies):
-            currencies.append((event_currency, event_currency))
-        self.currency.choices = sorted(currencies, key=lambda x: x[1].lower())
-
-    def validate_currency(self, field):
```

Comment:
Then you are supposed to disable the payment methods not compatible with your currency..